### PR TITLE
Spdm1.3 add initial support

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/DMTF/libspdm/issues/2296
Fix https://github.com/DMTF/libspdm/issues/2288

### CODE REVIEW

File: include/industry_standard/spdm.h [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 #pragma pack(1)
 
-/* 3 means SPDM spec 1.0, 1.1, 1.2 */
-#define SPDM_MAX_VERSION_COUNT 3
+/* 3 means SPDM spec 1.0, 1.1, 1.2, 1.3 */
```
Comment (steven-bellock) [inline include/industry_standard/spdm.h:17]: `3` -> `4`

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 2]
```diff
@@ -2649,13 +2649,15 @@ libspdm_return_t libspdm_init_context_with_secured_context(void *spdm_context,
         sizeof(context->transcript.message_m.buffer);
 #endif
     context->response_state = LIBSPDM_RESPONSE_STATE_NORMAL;
-    context->local_context.version.spdm_version_count = 3;
+    context->local_context.version.spdm_version_count = 4;
```
Comment (steven-bellock) [inline library/spdm_common_lib/libspdm_com_context_data.c:2652]: `4` -> `SPDM_MAX_VERSION_COUNT`

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 3]
```diff
@@ -2649,13 +2649,15 @@ libspdm_return_t libspdm_init_context_with_secured_context(void *spdm_context,
...
     context->local_context.version.spdm_version[0] = SPDM_MESSAGE_VERSION_10 <<
```
Comment (steven-bellock) [inline library/spdm_common_lib/libspdm_com_context_data.c:2653]: What does this actually do? Does this say the local endpoint supports all SPDM versions?

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/industry_standard/spdm.h (hunk 1 above)
Comment (jyao1) [reply include/industry_standard/spdm.h]: done

File: library/spdm_common_lib/libspdm_com_context_data.c (hunk 2 above)
Comment (jyao1) [reply library/spdm_common_lib/libspdm_com_context_data.c]: done

File: library/spdm_common_lib/libspdm_com_context_data.c (hunk 3 above)
Comment (jyao1) [reply library/spdm_common_lib/libspdm_com_context_data.c:2653]: Right.

Otherwise, there is no way to validate the 1.3 support.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**